src/firewall.c

```
@@ -244,6 +262,13 @@ fw_set_authservers(void)
 int
 fw_destroy(void)
 {
+    s_config *config = config_get_config();
+
+    if (config->bw_shaping) {
+        debug(LOG_INFO, "Removing Bandwidth Shaping settings");
+        bw_shaping_destroy();
```

There is also no error handling (e.g. logging) here, although it probably is less critical. It would be nice to generate a warning if e.g. the tc syntax changes. Right now, any errors would be hidden.